### ISSUE TITLE
libcontrol

### ISSUE THREAD

**Issue by adriangrigore** [9fans/plan9port#614]
Any reason `libcontrol` is not ported?

**Comment by noelhunt** [comment]
Yes, I have wondered that myself. I have actually built libcontrol, but
I would like to see examples of its use. I am certain I read somewhere
that there was only ever one program written using it, but I've never
found such a thing in any of the released Plan9 source.

**Comment by adriangrigore** [comment]
I compiled it successfully on OpenBSD an actually created a small program with it!

> I would like to see examples of its use. I am certain I read somewhere
that there was only ever one program written using it, but I've never
found such a thing in any of the released Plan9 source.

https://9p.io/sources/plan9/sys/src/cmd/auth/factotum/fgui.c

**Comment by noelhunt** [comment]
Many thanks, I will try this out.

When I was experimenting with libcontrol a couple of years ago,
all I could find was a program to display a 'slider'; I found this
in the contrib source. There was one other program which drew a
large window with a text entry line (rectangle) at the bottom.
Very reminiscent of pads(9)/pi(9) and Rob Pike's editor 'jim'.
I think commands typed in the text entry rectangle had their
output displayed in the large window above.

Your code looks a lot more interesting.

**Comment by halfwit** [comment]
/sys/src/cmd/bitsy/keyboard.c uses libcontrol. I wrote a bit with it for my personal things, and really enjoyed working in it, but it does feel somewhat unfinished.
